src/main/java/org/example/elements/Animal.java

```diff
@@ -1,30 +1,188 @@
 package org.example.elements;
 
+import org.example.utils.IPositionChangeObserver;
 import org.example.utils.MapDirection;
 import org.example.utils.Vector2D;
 
-import java.util.Map;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
 
 public class Animal extends AbstractMapElement{
 
     public int lifeDays;
-    public int power;
-    public int[] genom;
-    public MapDirection direction;
+    public int initialEnergy;
+    public int reproduceEnergy;
+    public int stuffedEnergy;
+    public int moveEnergy;
+    public int childrenNum;
+    public int genomeLength;
+    public Genome genome;
+    public boolean isAlive;
+    public MapDirection direction = MapDirection.NORTH;
+    private final ArrayList<IPositionChangeObserver> observers;
 
-    public Animal(Vector2D position) {
+    public Animal(Vector2D position, int genomeLength, int initialEnergy, int stuffedEnergy, int reproduceEnergy, int moveEnergy) {
+        // initial position for animal
         super(position);
-        this.direction = MapDirection.NORTH;
+        this.direction.randomDirection();
```

Comment:
O ile rozumiem idee, to to chyba powinno być przypisanie, coś w stylu

```java
this.direction = MapDirection.randomDirection();
```

wtedy `randomDirection()` mogłaby być metodą statyczną